sysid-application/src/main/native/cpp/Util.cpp

```diff
@@ -52,3 +54,21 @@ std::string sysid::GetAbbreviation(std::string_view unit) {
     throw std::runtime_error("Invalid Unit");
   }
 }
+
+void sysid::SaveFile(std::string_view contents, const fs::path& path) {
+  // Create the path if it doesn't already exist.
+  fs::create_directories(path.root_directory());
+
+  // Open a fd_ostream to write to file.
+  std::error_code ec;
+  wpi::raw_fd_ostream ostream{path.string(), ec};
+
+  // Check error code.
+  if (ec) {
+    throw std::runtime_error("Cannot write to file: " + ec.message());
+  }
+
+  // Write contents.
+  ostream << contents;
+  ostream.flush();
```

Comment:
The destructor automatically flushes the file, so the call to flush here is redundant.
```cpp
raw_fd_ostream::~raw_fd_ostream() {
  if (FD >= 0) {
    flush();
    if (ShouldClose && ::close(FD) < 0)
      error_detected(std::error_code(errno, std::generic_category()));
  }
```